bodhi/tests/server/services/test_updates.py

```diff
@@ -5028,3 +5028,169 @@ def test_unexpected_exception(self, log_exception, waive_test_results, *args):
+        nvr = u'bodhi-2.0-1.fc17'
+        greenwave_api_post.return_value = {
+            u'unsatisfied_requirements': [
+                {
+                    u'item': {
+                        u'item': u'bodhi-2.0-1.fc17',
+                        u'type': u'koji_build'
+                    },
+                    u'scenario': None,
+                    u'testcase': u'dist.rpmdeplint',
+                    u'type': u'test-result-missing'
+                }
+            ],
+        }
+
+        up = self.db.query(Update).filter_by(title=nvr).one()
+        up.test_gating_status = TestGatingStatus.failed
+
+        post_data = dict(update=nvr, comment="This is expected", csrf_token=self.get_csrf_token())
+        res = self.app.post_json('/updates/%s/waive-test-results' % str(nvr), post_data, status=200)
+
+        greenwave_api_post.assert_called_once_with(
+            'https://greenwave-web-greenwave.app.os.fedoraproject.org/api/v1.0/decision',
+            {
+                'product_version': u'fedora-17',
+                'decision_context': u'bodhi_update_push_testing',
+                'subject': [
+                    {'item': u'bodhi-2.0-1.fc17', 'type': 'koji_build'},
+                    {'original_spec_nvr': u'bodhi-2.0-1.fc17'},
+                    {'item': u'FEDORA-2018-a3bbe1a8f2', 'type': 'bodhi_update'}
+                ]
+            }
+        )
+
+        waiverdb_api_post.assert_called_once_with(
+            'https://waiverdb-web-waiverdb.app.os.fedoraproject.org/api/v1.0/waivers/',
+            {
+                'username': u'guest',
+                'comment': u'This is expected',
+                'waived': True,
+                'product_version': u'fedora-17',
+                'testcase': u'dist.rpmdeplint',
+                'subject': {
+                    u'item': u'bodhi-2.0-1.fc17', u'type': u'koji_build'
+                }
+            }
+        )
+
+        self.assertEquals(list(res.json_body.keys()), ['update'])
+        self.assertEquals(
+            sorted(res.json_body['update'].keys()),
+            [
+                u'alias', u'autokarma', u'bugs', u'builds', u'close_bugs', u'comments', u'compose',
+                u'content_type', u'critpath', u'date_approved', u'date_modified', u'date_pushed',
+                u'date_stable', u'date_submitted', u'date_testing', u'display_name',
+                u'greenwave_summary_string', u'greenwave_unsatisfied_requirements', u'karma',
+                u'locked', u'meets_testing_requirements', u'notes', u'old_updateid', u'pushed',
+                u'release', u'request', u'require_bugs', u'require_testcases', u'requirements',
+                u'severity', u'stable_karma', u'status', u'submitter', u'suggest', u'test_cases',
+                u'test_gating_status', u'title', u'type', u'unstable_karma', u'updateid', u'url',
+                u'user'
+            ]
+        )
```

Comment:
It might be easier and more thorough to do this as an assertion in place of the above two:

```
self.assertEqual(res.json_body, up.__json__())
```

Reply:
+1

Reply:
So this unfortunately doesn't work as it seems one of them include openid and avatar information while the other doesn't :(
````
E       -                 u'user': {u'avatar': u'https://seccdn.libravatar.org/avatar/eb48e08cc23bcd5961de9541ba5156c385cd39799e1dbf511477aa4d4d3a37e7?s=24&d=retro',
E       +                'user': {'avatar': None,
````

````
E       -                           u'openid': u'guest.id.fedoraproject.org',
E       +                         'openid': None,
````

I'll look if I can fix this but I'm not seeing anywhere else where you used this assertion before so seems to be a new territory

Reply:
Some mock magic fixed it :)

````
@mock.patch('bodhi.server.models.User.openid', mock.MagicMock(return_value=None))
@mock.patch('bodhi.server.models.User.avatar', mock.MagicMock(return_value=None))
````